src/explore-education-statistics-frontend/src/components/PrintThisPageAnalytics.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import classNames from 'classnames';
+import { logEvent } from '@frontend/services/googleAnalyticsService';
+
+const PrintThisPage = () => {
+  return (
+    <div className={classNames('govuk-!-margin-top-6', 'dfe-print-hidden')}>
+      <a
+        href="#"
+        onClick={() => {
+          window.print();
+          logEvent(
+            'Page print',
+            'Print this page link selected',
+            window.location.pathname,
```

Comment:
I think the order of the parameters here may need checking.

Wanna keep the organisation of the event logging consistent, `(category, action, label)` - other instances of `logEvent` have used the pathname as the category, here you've put it as the label

Reply:
Hey @bennettstuart,

I'm not 100% convinced that using page URLs as Categories is the right usage of Category - I would have thought that a Category would be something like "Search" or "Table Creation" and under those you'd have actions like "Search performed" and labels like "Searching from header search bar on page <url>".

Having said this, my only experience of this is from SPRINT which is how I set up the GA stuff there, so I am heavily biased that way, and will be happily proven wrong :)

Cheers,
Duncan

Reply:
I'm happy to go with any method, so long as it's consistent.

I thought it might be nice to look at analytic events within a page as a whole, which is why I was categorising by the path. Especially when a common component's context may change slightly dependant on what the contents of the page they appear in.

It's a shame there isn't a 4th property we could use to organise events.

I think the `label` prop should hold the event's _value_, where applicable. For instance, on the PageSearchFormWithAnalytics component we track the search-term the user entered against the label property of the event.

I propose something like this:

- Category - What type of event?
  - e.g. "Search" or "Table Creation" 
- Action - What happened?
  - e.g. "Search performed" or "Table Created"
- Labels -  What was the value entered? 
  - e.g. "\<Whatever the user searched\>" or "\<the created table configuration\>".

-----

Perhaps we could append the path to the end of the action property to give something like

> Page Search  -  Search Performed /Downloads  -  "latest student absences"

I'd like to hear your thoughts on this @duncan-at-hiveit 

@HiveIT-Lewis we'll need to update all the logEvents to use the same format, once we have it nailed down

Reply:
Hey @bennettstuart,

I honestly don't know what the general convention is, but I'd also advise you to take a look at Goals in GA, and how they can be linked to Events - I certainly know that you can link Events to Goals by their Category - can't 100% remember if you can by their actions as well, but if so, you'd then need to start using regex to match on a dynamic Action name.

While you're talking about search and GA, don't miss the trick that we missed on SPRINT (and are subsequently going back to address), which is that GA has a special notion of search tracking that is different from event tracking, so it's better to log search actions using GA's search features rather than its events features.

Reply:
Hi @HiveIT-Lewis & @duncan-at-hiveit ,

from [analyticsjs/events](https://developers.google.com/analytics/devguides/collection/analyticsjs/events)

Field Name | Value Type | Required | Description
-- | -- | -- | --
eventCategory | text | yes | Typically the object that was interacted with (e.g. 'Video')
eventAction | text | yes | The type of interaction (e.g. 'play')
eventLabel | text | no | Useful for categorizing events (e.g. 'Fall Campaign')

Maybe using the label for the path is the best solution, if we're going to log that page the event occurred on.

Thanks,
Stu